models/work-item-link-category-repository.go

```diff
@@ -0,0 +1,169 @@
+package models
+
+import (
+	"log"
+
+	"golang.org/x/net/context"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/jinzhu/gorm"
+	satoriuuid "github.com/satori/go.uuid"
+)
+
+// NewWorkItemLinkCategoryRepository creates a work item link category repository based on gorm
+func NewWorkItemLinkCategoryRepository(db *gorm.DB) *GormWorkItemLinkCategoryRepository {
+	return &GormWorkItemLinkCategoryRepository{db}
+}
+
+// GormWorkItemLinkCategoryRepository implements WorkItemLinkCategoryRepository using gorm
+type GormWorkItemLinkCategoryRepository struct {
+	db *gorm.DB
+}
+
+// Create creates a new work item link category in the repository.
+// Returns BadParameterError, ConversionError or InternalError
+func (r *GormWorkItemLinkCategoryRepository) Create(ctx context.Context, name *string, description *string) (*app.WorkItemLinkCategory, error) {
+	if name == nil || *name == "" {
+		return nil, NewBadParameterError("name", name)
+	}
+	created := WorkItemLinkCategory{
+		// Omit "lifecycle" and "ID" fields as they will be filled by the DB
+		Name:        *name,
+		Description: description,
+	}
+	db := r.db.Create(&created)
+	if db.Error != nil {
+		return nil, NewInternalError(db.Error.Error())
+	}
+	// Convert the created link category entry into a JSONAPI response
+	result := ConvertLinkCategoryFromModel(&created)
```

Comment:
I don't want to make this a member because eventually, all the knowledge about the `app.*` namespace shall be moved out of the `model` package. And then the domain layer (probably `app`) needs to do the conversion.